src/libraries/Microsoft.PowerFx.Core/Public/Values/TableValue.cs

```diff
@@ -85,16 +87,40 @@ private static ErrorValue ArgumentOutOfRange(IRContext irContext)
         // Async because derived classes may back this with a network call. 
         public virtual async Task<DValue<RecordValue>> AppendAsync(RecordValue record)
         {
-            // fails by default
-            throw new System.NotImplementedException("It is not possible to append to a TableValue directly.");
+            throw ThrowNotImplementedException(MethodBase.GetCurrentMethod().Name);
+        }
+
+        public virtual async Task<DValue<RecordValue>> PatchCoreAsync(RecordValue originalRecord, RecordValue newRecord)
+        {
+            throw ThrowNotImplementedException(MethodBase.GetCurrentMethod().Name);
+        }
+
+        public async Task<DValue<RecordValue>> PatchAsync(RecordValue originalRecord, RecordValue newRecord)
+        {
+            var recordType = ((TableType)IRContext.ResultType).ToRecord();
+
+            // Resolve from display names to logical names, if any.
+            var resolvedOriginalRecord = recordType.ResolveToLogicalNames(originalRecord);
+            var resolvedNewRecord = recordType.ResolveToLogicalNames(newRecord);
+
+            return await PatchCoreAsync(resolvedOriginalRecord, resolvedNewRecord);
         }
 
         // Return boolean value
         // Async because derived classes may back this with a network call. 
         public virtual async Task<DValue<BooleanValue>> RemoveAsync(RecordValue record)
         {
-            // fails by default
-            throw new System.NotImplementedException("It is not possible to remove from a TableValue directly.");
+            throw ThrowNotImplementedException(MethodBase.GetCurrentMethod().Name);
+        }
+
+        public virtual bool TryFindRecord(RecordValue originalRecord, out int index)
+        {
+            throw ThrowNotImplementedException(MethodBase.GetCurrentMethod().Name);
+        }
+
+        private Exception ThrowNotImplementedException(string methodName)
+        {
+            return new NotImplementedException($"It is not possible to append to call {methodName} method from TableValue directly.");
         }
```

Comment:
actually, I think all fo these cases should return a runtime error (not throw an exception).